src/main/java/hudson/remoting/Channel.java

```diff
@@ -270,7 +274,9 @@
      * without telling us anything, the {@link SocketOutputStream#write(int)} will
      * return right away, and the socket only really times out after 10s of minutes.
      */
-    private volatile long lastHeard;
+    @GuardedBy("lastHeardLock")
+    private long lastHeard;
+    private Object lastHeardLock = new String("lastHeardLock");
```

Comment:
I think an actual `Lock` and `Condition` would be better. In either case locking on a String is a bad idea so even if you want to lock on an object's built in syncronizer it should be `private final Object lastHeardLock = new Object()`